### DISCUSSION TITLE
load last session at startup/automate session loading

### DISCUSSION THREAD

**Discussion by Carnarts** [DISTRHO/Cardinal#708] [General]
I'm trying to find a way for autoloading a session when Cardinal starts up, defaulting to the last session. Running on Raspberry pi 5, patchbox OS. Is there a way to do this? I've been trying with both Carla and Cardinal and both don't seem to be able to do this easily, if at all. Many thanks!

**Comment by falkTX** [comment]
not really a feature of the standalone, but since you mention Carla you can save a project and pass it as CLI argument next time you start it up again.

**Comment by Carnarts** [comment]
Thanks for the reply. Unfortunately with Carla that still doesn't work for me. I can start Carla up like that but if I try and start it with a project file via the CLI, Carla says the file is unreadable or doesn't exist after Carla runs.  If I load the project through the gui it's fine though. I doublechecked the paths for the cli command is correct and it's not that.

**Comment by Carnarts** [comment]
Ok I managed to do it after a lot of messing around!
It didn't work properly for me as a service as the audio from my soundcard didn't work that way, but this works for me:

Steps to Auto-Launch Cardinal at Boot:
Create the autostart desktop entry:

Create a folder if it doesn't exist:
`mkdir -p ~/.config/autostart`

Create and edit launch_cardinal.desktop in ~/.config/autostart:
`nano ~/.config/autostart/launch_cardinal.desktop`

Add the following content to launch_cardinal.desktop:

```
[Desktop Entry]
Type=Application
Exec=/home/patch/launch_cardinal.sh
Hidden=false
NoDisplay=false
X-GNOME-Autostart-enabled=true
Name=Launch Cardinal
Comment=Start Cardinal at login
```

Create the launch_cardinal.sh script:

Create and edit launch_cardinal.sh:
`nano /home/patch/launch_cardinal.sh`

Add the following content to launch_cardinal.sh:

```
#!/bin/bash
lxterminal -e /home/patch/start_cardinal_terminal.sh
```

Create the start_cardinal_terminal.sh script:

Create and edit start_cardinal_terminal.sh:

`nano /home/patch/start_cardinal_terminal.sh`

Add the following content to start_cardinal_terminal.sh:

```
#!/bin/bash
export DISPLAY=:0
/usr/bin/CardinalNative
```

Make both scripts executable:

`chmod +x /home/patch/launch_cardinal.sh /home/patch/start_cardinal_terminal.sh
`
Result:
At boot, the Raspberry Pi will automatically open a terminal and run Cardinal. If you write over the default template with your current patch it will load that :)

**Comment by Carnarts** [comment]
Another question is, is there a way of editing default settings for Cardinal? eg starting with a predefined buffer size, OSC on, Audio input enabled etc?

**Reply by Carnarts** [reply]
It would great as a feature request to have options for Cardinal standalone to start with these features, eg OSC enabled etc :)
